tests/foreman/ui_airgun/test_contenthost.py

```diff
@@ -14,103 +14,502 @@
+        2. Register the host to activation key
+        3. Install the package that has errata
+        4. Refresh content host subscription running:
+            "subscription-manager refresh  && yum repolist"
+        5. Ensure errata is available for installation
+        6. Refresh content host subscription running:
+            "subscription-manager refresh  && yum repolist"
+
+    :expectedresults: errata is available in installable errata list
+
+    :BZ: 1463818
+
+    :CaseLevel: System
+    """
+    vm.run('yum install -y {0}'.format(FAKE_1_CUSTOM_PACKAGE))
+    result = vm.run('rpm -q {0}'.format(FAKE_1_CUSTOM_PACKAGE))
+    assert result.return_code == 0
+    result = vm.run('subscription-manager refresh  && yum repolist')
+    assert result.return_code == 0
+    with session:
+        chost = session.contenthost.read(vm.hostname)
+        assert FAKE_2_ERRATA_ID in {errata['Id'] for errata in chost['errata']['table']}
+        result = vm.run('subscription-manager refresh  && yum repolist')
+        assert result.return_code == 0
+        chost = session.contenthost.read(vm.hostname)
+        assert FAKE_2_ERRATA_ID in {errata['Id'] for errata in chost['errata']['table']}
+
+
+@tier3
+@upgrade
+def test_positive_check_ignore_facts_os_setting(session, vm, module_org, request):
+    """Verify that 'Ignore facts for operating system' setting works
+    properly
+
+    :steps:
+
+        1. Create a new host entry using content host self registration
+           procedure
+        2. Check that there is a new setting added "Ignore facts for
+           operating system", and set it to true.
+        3. Upload the facts that were read from initial host, but with a
+           change in all the operating system fields to a different OS or
+           version.
+        4. Verify that the host OS isn't updated.
+        5. Set the setting in step 2 to false.
+        6. Upload same modified facts from step 3.
+        7. Verify that the host OS is updated.
+        8. Verify that new OS is created
+
+    :id: 71bed439-105c-4e87-baae-738379d055fb
+
+    :customerscenario: true
+
+    :expectedresults: Host facts impact its own values properly according
+        to the setting values
+
+    :BZ: 1155704
+
+    :CaseLevel: System
+    """
+    major = str(gen_integer(15, 99))
+    minor = str(gen_integer(1, 9))
+    expected_os = "RedHat {}.{}".format(major, minor)
+    set_ignore_facts_for_os(False)
+    host = entities.Host().search(query={
+        'search': 'name={0} and organization_id={1}'.format(
+            vm.hostname, module_org.id)
+    })[0].read()
+    with session:
+        # Get host current operating system value
+        os = session.contenthost.read(vm.hostname)['details']['os']
+        # Change necessary setting to true
+        set_ignore_facts_for_os(True)
+        # Add cleanup function to roll back setting to default value
+        request.addfinalizer(set_ignore_facts_for_os)
```

Comment:
Think that this is a candidate for a fixture

Reply:
but we can ignor this if it's used only here

Reply:
It's being used only here